core/src/main/scala/org/apache/spark/scheduler/SchedulableBuilder.scala

```diff
@@ -69,60 +72,81 @@ private[spark] class FairSchedulableBuilder(val rootPool: Pool, conf: SparkConf)
   val DEFAULT_WEIGHT = 1
 
   override def buildPools() {
-    var is: Option[InputStream] = None
+    var fileData: Option[FileData] = None
     try {
-      is = Option {
-        schedulerAllocFile.map { f =>
-          new FileInputStream(f)
-        }.getOrElse {
-          Utils.getSparkClassLoader.getResourceAsStream(DEFAULT_SCHEDULER_FILE)
-        }
-      }
-
-      is.foreach { i => buildFairSchedulerPool(i) }
+      fileData = getFileData()
+      fileData.foreach { data => buildFairSchedulerPool(data) }
+    } catch {
+      case NonFatal(t) =>
+        logError("Error while building the fair scheduler pools: ", t)
+        throw t
     } finally {
-      is.foreach(_.close())
+      fileData.foreach(_.inputStream.close())
     }
 
     // finally create "default" pool
     buildDefaultPool()
   }
 
+  private def getFileData(): Option[FileData] = {
```

Comment:
Can you in-line this like it was before?  It's not very complicated to easier to just be in-line in the buildPools method.  Also, the old code structure simplified the Option creation.